client/src/main/java/org/apache/rocketmq/client/impl/producer/DefaultMQProducerImpl.java

```diff
@@ -713,15 +713,25 @@ private TopicPublishInfo tryToFindTopicPublishInfo(final String topic) {
         TopicPublishInfo topicPublishInfo = this.topicPublishInfoTable.get(topic);
         if (null == topicPublishInfo || !topicPublishInfo.ok()) {
             this.topicPublishInfoTable.putIfAbsent(topic, new TopicPublishInfo());
-            this.mQClientFactory.updateTopicRouteInfoFromNameServer(topic);
             topicPublishInfo = this.topicPublishInfoTable.get(topic);
+            if (!topicPublishInfo.checkNotFoundFlag()) {
+                this.mQClientFactory.updateTopicRouteInfoFromNameServer(topic);
+                topicPublishInfo = this.topicPublishInfoTable.get(topic);
+            }
         }
 
         if (topicPublishInfo.isHaveTopicRouterInfo() || topicPublishInfo.ok()) {
             return topicPublishInfo;
         } else {
-            this.mQClientFactory.updateTopicRouteInfoFromNameServer(topic, true, this.defaultMQProducer);
-            topicPublishInfo = this.topicPublishInfoTable.get(topic);
+            boolean found;
+            if (!topicPublishInfo.checkNotFoundFlag()) {
+                found = this.mQClientFactory.updateTopicRouteInfoFromNameServer(topic, true, this.defaultMQProducer);
```

Comment:
updateTopicRouteInfoFromNameServer return to whether topicRouteInfo is changed, not whether the route is found

Reply:
(line 723)
`if (topicPublishInfo.isHaveTopicRouterInfo() || topicPublishInfo.ok())`
When it is false, if topicRouteInfo does not change, it means that it is not found

Reply:
(MQClientInstance.updateTopicRouteInfoFromNameServer, line624)
```
if (topicRouteData != null) {
  TopicRouteData old = this.topicRouteTable.get(topic);
  boolean changed = topicRouteDataIsChange(old, topicRouteData);
  ...
  if (changed) {
    ...
    return true
  }
  ...
}
return false;     
```
1、TBW102 not found. Return false because the topicRouteData of TBW102 is null.
2、TBW102 found. Return true because it is different from topicRouteData of the origin topic.